src/main/java/ch/njol/skript/expressions/ExprDateAgoLater.java

```diff
@@ -45,28 +45,37 @@
                 "%timespan% (later|(from|after) [the] [date] %-date%)");
     }
 
+    @Override
+    @SuppressWarnings({"unchecked", "null"})
+    public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+        timespan = (Expression<Timespan>) exprs[0];
+        date = (Expression<Date>) exprs[1];
+        ago = matchedPattern == 0;
+        return true;
+    }
+
     @SuppressWarnings("null")
     private Expression<Timespan> timespan;
     @Nullable
     private Expression<Date> date;
 
     private boolean ago;
 
-    @SuppressWarnings("null")
-    @Nullable
     @Override
+    @Nullable
+    @SuppressWarnings("null")
     protected Date[] get(Event e) {
+		if (this.date == null || this.date.getSingle(e) == null || this.timespan == null || this.timespan.getSingle(e) == null)
```

Comment:
I would avoid calling getSingle twice. Perhaps just store the value from getSingle and check whether that is null

Reply:
Sure, is this needed btw `(this.date == null || this.timespan == null)`?

Reply:
date yes, timespan no, see the pattern: the date expression is optional, the timespan isn't